src/main/java/com/example/bookshop/service/impl/BookServiceImpl.java

```diff
@@ -47,4 +54,28 @@ public BookDto findById(Long id) {
                 .orElseThrow(() -> new EntityNotFoundException("Can't find book"
                         + " with this id: " + id));
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(CreateBookRequestDto createBookRequestDto, Long id) {
+        if (!bookRepository.existsById(id)) {
+            throw new EntityNotFoundException("There is no book with such id: " + id);
+        }
+        Book book = bookMapper.toBook(createBookRequestDto);
+        book.setId(id);
+        return bookMapper.toDto(bookRepository.save(book));

```

Comment:
this is a bit wrong

you should first find a book by id

then update it with the help of Mapstruct:
```
void updateBook(CreateBookRequestDto dto, @MappingTarget Book book);
```

and then save


it will look like (pseudocode):
```
Book book = repo.findById(id).orElseThrow(...);
mapper.updateBook(dto, book);
return mapper.toDto(repo.save(book));
```

and you can remove the exists check as well